src/module.c

```diff
@@ -188,6 +188,12 @@ static int parseRangeArguments(RedisModuleCtx *ctx, Series *series, int start_in
 static int parseCountArgument(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, long long *count) {
     int offset = RMUtil_ArgIndex("COUNT", argv, argc);
     if (offset > 0) {
+        if (strcasecmp(RedisModule_StringPtrLen(argv[offset - 1], NULL), "AGGREGATION") == 0) {
+            if (strcasecmp(RedisModule_StringPtrLen(argv[offset + 2], NULL), "COUNT") != 0) {
```

Comment:
@danni-m @ashtul it's future risky to have this offset+2 check, it means that in the future when we'll had more flags it will break. The offset+2 assumes that the only flag we can have after the AGGREGATION is COUNT and it's only correct till we'll add another flad.

Reply:
Reverted